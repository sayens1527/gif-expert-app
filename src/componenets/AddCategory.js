import React , {useState} from 'react'
import PropTypes from 'prop-types'

const AddCategory = ( { setCategories } ) => {

    let [newCategory, setNewCategory] = useState('') ;

    const handleChange = (e) => {
        setNewCategory(e.target.value);
    };

    const onSubmit = (e) => {
        e.preventDefault();
        if(newCategory.length > 2){
            setCategories(categories => [newCategory, ...categories]);
            setNewCategory('')
        }
        
    }

    return (
        <form onSubmit = { onSubmit } >
            <h2>Add category</h2>
            <input 
                type='text' 
                value={newCategory}
                onChange = { handleChange }
            >
            </input>
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory
