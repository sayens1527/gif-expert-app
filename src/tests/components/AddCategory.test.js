import React from 'react';
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';
import '@testing-library/jest-dom'


describe('Test in AddCategory component', () => {

    const setCategories = jest.fn();
    let wraper = shallow(<AddCategory setCategories={ setCategories }></AddCategory>);

    beforeEach( () => { 
        jest.clearAllMocks();
        wraper = shallow(<AddCategory setCategories={ setCategories }></AddCategory>);
    });

    test('should match with snapshot', () => {
    
        expect(wraper).toMatchSnapshot();

    });

    test('should change the iput text change', () => {

        const input = wraper.find('input');

        input.simulate('change', {target:{value:'Hola mundo'}});

        //const inputValue = input.prop('value');

        //expect(inputValue).toBe('Hola mundo');
        
    });

    test('should not call the set category function', () => {
        

        const form = wraper.find('form');

        form.simulate('submit', { preventDefault(){} } );

        expect(setCategories).not.toHaveBeenCalled();

    });

    test('should call the set category function', () => {
        
        const input = wraper.find('input');

        input.simulate('change', {target:{value:'Hola mundo'}});

        const form = wraper.find('form');

        form.simulate('submit', { preventDefault(){} } );

        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

        const inputAfterSubmit = wraper.find('input');
        const inputValueAfterSubmit = inputAfterSubmit.prop('value')

        expect(inputValueAfterSubmit).toBe('');

    });
    
    
    
})

