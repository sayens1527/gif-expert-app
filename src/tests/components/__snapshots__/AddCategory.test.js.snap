// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in AddCategory component should match with snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    Add category
  </h2>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
